### PR TITLE
i04 thawing plan

### LINKED ISSUE #102: Write i04 thawing plan
i04 would like a bluesky plan that:
* Takes a time for how long to spend thawing
* Turns on the thawer
* Sets the smargon velocity such that a full rotation takes that long
* Turns off thawing

### PR DESCRIPTION
**Pull request by DominicOram**
Fixes #102

### CODE REVIEW

File: src/mx_bluesky/i04/thawing_plan.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from dodal.common import inject
+from dodal.devices.smargon import Smargon
+from dodal.devices.thawer import Thawer, ThawerStates
+
+
+def thaw(
+    time_to_thaw: float,
+    rotation: float = 360,
+    thawer: Thawer = inject("thawer"),  # type: ignore
+    smargon: Smargon = inject("smargon"),  # type: ignore
```
Comment (callumforrester) [inline src/mx_bluesky/i04/thawing_plan.py:13]: Did the type checker not like this? That's worrying...
Any ideas @DiamondJoseph

File: .vscode/settings.json
Comment (callumforrester) [inline .vscode/settings.json:1]: Nit: I think this is better in a separate PR

**Review by callumforrester — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -22,6 +22,7 @@ dependencies = [
     "opencv-python",
     "pydantic",
     "dls-dodal @ git+https://github.com/DiamondLightSource/dodal.git@main",
+    "blueapi>=0.4.3-a3",
```
Comment (DiamondJoseph) [inline pyproject.toml:25]: Shouldn't be required after get MsgGenerator from dodal

File: src/mx_bluesky/i04/thawing_plan.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import bluesky.plan_stubs as bps
+import bluesky.preprocessors as bpp
+from blueapi.core import MsgGenerator
+from dodal.common import inject
```
Comment (DiamondJoseph) [inline src/mx_bluesky/i04/thawing_plan.py:4]:
```suggestion
from dodal.common import MsgGenerator, inject
```

File: src/mx_bluesky/i04/thawing_plan.py (hunk 1 above)
Comment (DominicOram) [reply src/mx_bluesky/i04/thawing_plan.py:13]: `inject` returns a string. I thought about making an issue but I'm not sure how you could fix it cleanly.

File: .vscode/settings.json
Comment (DominicOram) [reply .vscode/settings.json:1]: Yh, probably, this was me being lazy.

File: src/mx_bluesky/i04/thawing_plan.py (hunk 1 above)
Comment (DiamondJoseph) [reply src/mx_bluesky/i04/thawing_plan.py:13]: Looks like mypy is still happy with it, but Pylance in the IDE isn't. If inject gains `-> Any` it behaves again. I'll make a PR to dodal.

**Review by DiamondJoseph — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
Comment (DominicOram) [reply .vscode/settings.json:1]: On second thoughts I'm sticking by my laziness

**Review by DiamondJoseph — APPROVED** [review]
I think it needs __init__.py in i04?

**Pull request merged**